### DISCUSSION TITLE
Default selection in MultiItemSelector

### DISCUSSION THREAD

**Discussion by Drevsh** [spring-projects/spring-shell#467] [Ideas]
I have a usecase where I want to offer multi selection with some values already pre selected. Afaik this is currently not supported right?

The only pre selection, which I can see, is a single selection via the `void org.springframework.shell.component.support.AbstractSelectorComponent.setDefaultExpose(I defaultExpose)`. But this is only useful for the SingleItemSelector. 

Do you think this use cases should be supported? I think it would make sense to either extend the  `setDefaultExpose` to also support collections, or rather to have this information directly on the SelectorItem/SelectorItemWrapper. So in addition to `enabled` there could be another boolean field `selected`. 

During the building of the ItemStateView this could then be extracted and the item could be selected.

I also think `void org.springframework.shell.component.support.AbstractSelectorComponent.initialExpose(C context)`/`setDefaultExpose` would then be obsolete.

**Comment by Drevsh** [comment]
I implemented the changes here:
https://github.com/Drevsh/spring-shell/commit/b5e26ab470c8a015a9da9d651e24a31e1292e991 

I think it works, but I'm a bit unhappy about the constructor since overloading is now not possible. Maybe a builder pattern is now needed?

**Reply by jvalkeal** [reply]
Thx, do you want to PR these changes as it's then easier to try and comment. 

I've also not been happy about `AbstractSelectorComponent` as it was created to have some shared features in one place but then single/multi have their own features. Probably one area to refactor with `3.x`. Most likely reason why I didn't implement pre-selection was to avoid bad behaviour or bugs as it has no meaning with single selector.

**Reply by Drevsh** [reply]
Created a pull request. Since this is my first time contributing I hope there are not that many problems.
